ArcGISRuntimeSDKQt_CppSamples/Geometry/FormatCoordinates/FormatCoordinates.cpp

```diff
@@ -0,0 +1,163 @@
+// limitations under the License.
+// [Legal]
+
+#include "FormatCoordinates.h"
+
+#include "Basemap.h"
+#include "CoordinateFormatter.h"
+#include "Graphic.h"
+#include "Map.h"
+#include "MapQuickView.h"
+#include "Point.h"
+#include "SimpleMarkerSymbol.h"
+
+#include <QDebug>
+
+using namespace Esri::ArcGISRuntime;
+
+FormatCoordinates::FormatCoordinates(QQuickItem* parent) :
+    QQuickItem(parent)
+{
+}
+
+FormatCoordinates::~FormatCoordinates()
+{
+}
+
+void FormatCoordinates::init()
+{
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<FormatCoordinates>("Esri.Samples", 1, 0, "FormatCoordinatesSample");
+}
+
+void FormatCoordinates::componentComplete()
+{
+    QQuickItem::componentComplete();
+
+    // find QML MapView component
+    m_mapView = findChild<MapQuickView*>("mapView");
+
+    // create a new basemap instance
+    Basemap* basemap = Basemap::imagery(this);
+
+    // create a new map instance
+    m_map = new Map(basemap, this);
+
+    // create a graphic
+    Esri::ArcGISRuntime::Point geometry(m_startLongitude, m_startLatitude, SpatialReference::wgs84());
+    SimpleMarkerSymbol* symbol = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle::X, QColor(Qt::red), 15.0, this);
+    Graphic* graphic = new Graphic(geometry, symbol, this);
+
+    // create a new graphics overlay instance
+    // and add insert the graphic
+    GraphicsOverlay* go = new GraphicsOverlay(this);
+    go->graphics()->append(graphic);
+
+    // set map and graphics overlay on the map view
+    m_mapView->setMap(m_map);
+    m_mapView->graphicsOverlays()->append(go);
+
+    connectSignals();
+
+    // Set initial position to Building Q
+    handleTextUpdate("Decimal Degrees", "34.056195N 117.195723W");
+}
+
+void FormatCoordinates::connectSignals()
+{
+  // connect to the mouse clicked signal on the MapQuickView
+  connect(m_mapView, &MapQuickView::mouseClicked, this, [this](QMouseEvent& mouseEvent)
+  {
+    // get the point from the mouse point
+    Esri::ArcGISRuntime::Point mapPoint = m_mapView->screenToLocation(mouseEvent.x(), mouseEvent.y());
+
+    // using the point, refresh the graphic and the text
+    handleLocationUpdate(mapPoint);
+  });
+}
+
+// handle case where the user changed one of the text fields
+void FormatCoordinates::handleTextUpdate(QString textType, QString text)
+{
+  Esri::ArcGISRuntime::Point point = createPointFromText(textType, text);
+  handleLocationUpdate(point);
+}
+
+// handle case where user clicked on the map
+void FormatCoordinates::handleLocationUpdate(Esri::ArcGISRuntime::Point point)
+{
+  if (!point.isEmpty())
+  {
+    m_mapView->graphicsOverlays()->at(0)->graphics()->at(0)->setGeometry(point);
+    setTextFromPoint(point);
+  }
+}
+
+Esri::ArcGISRuntime::Point FormatCoordinates::createPointFromText(QString textType, QString text)
+{
+  //! [FormatCoordinates CoordinateFormatter various text to point]
+  if ("Decimal Degrees" == textType
```

Comment:
Optional, but it might be cleaner to make an enum for the different coordinate types instead of relying on strings here and in your QML code. Another option could be to define static strings, and use getters to access the string, so that you don't have the string spelled out in multiple places.